src/curl.c

```diff
@@ -185,10 +199,7 @@ curl_multi_sock_cb(CURL *easy, curl_socket_t fd, int what, void *envp,
 
 	if (what == CURL_POLL_REMOVE) {
 		say_debug("curl %p: remove fd=%d", env, fd);
-		assert(env->watcher != NULL);
```

Comment:
Old logic expected that watcher always exists. But after your change absence of watcher will be ignored. Is it expected?

Reply:
Seems it was lost. Now, function's body is exactly the same as a condition's body.